### PR TITLE
[docs] [Integrations] Docs added for Spark YSQL integrations

### LINKED ISSUE #13461: [Docs] [Integration] [YSQL] Spark Integration with YugabyteDB
### Description

This Issue is to document the usage of Spark SQL API's with YugabyteDB. The documentation for that is [here](https://docs.google.com/document/d/1cjANTuKFLntfiCs8eq2JWvI8gmvtREawpQmJrEXC8Ho/edit?usp=sharing). It should go under integration page in Apache Spark for [YSQL - Scala], [YSQL - Python] and [Spark-SQL] tabs [here](https://docs.yugabyte.com/preview/integrations/apache-spark/java/).

### PR DESCRIPTION
**Pull request by aishwarya24**
resolves #13461 

@netlify /preview/integrations/apache-spark/java-ysql/

### CODE REVIEW

**Review by priyanshi-yb — APPROVED** [review]

File: docs/content/preview/integrations/apache-spark/python-ysql.md [hunk 1]
```diff
@@ -0,0 +1,292 @@
...
+- `partitionColumn` - the column on the basis of which a partition occurs.
+
+These options help in breaking down the whole task into `numPartitions` parallel tasks on the basis of the `partitionColumn`, with the help of minimum and maximum value of the column.
+
+Additionally, the following two options help in optimizing the SQL queries executing on this dataframe if those SQL queries consist of some filters or aggregate functions by pushing down those filters and aggregates to the YugabyteDB through the JDBC connector.
+
+- `pushDownPredicate` - optimizes the query by pushing down the filters to YugabyteDB through the JDBC connector.
+- `pushDownAggregate` - optimizes the query by pushing down the aggregated to YugabyteDB through the JDBC connector.
+
+```spark
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/python-ysql.md:269]: For example, the following [whatever it is and whatever it does]:

File: docs/content/preview/integrations/apache-spark/python-ysql.md [hunk 2]
```diff
@@ -0,0 +1,292 @@
...
++---------+
+|sum(ceil)|
++---------+
+|1049414.0|
++---------+
+```
+
+### Verify parallelism
+
+To verify that the Spark job is created,
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/python-ysql.md:286]:
```suggestion
To verify that the Spark job is created, do the following:
```

File: docs/content/preview/integrations/apache-spark/python-ysql.md [hunk 3]
```diff
@@ -0,0 +1,292 @@
...
+
+1. Navigate to the Spark UI using <https://localhost:4040>. If your port 4040 is in use, then change the port to the one mentioned in the output of your [`pyspark`](#start-python-spark-shell-with-yugabytedb-driver) shell.
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/python-ysql.md:288]: the one mentioned when you started the [`pyspark`](#start-python-spark-shell-with-yugabytedb-driver) shell.

File: docs/content/preview/integrations/apache-spark/scala-ysql.md [hunk 4]
```diff
@@ -0,0 +1,313 @@
...
+
+</ul>
+
+The following tutorial describes how to use Scala's Spark API [`spark-shell`](https://spark.apache.org/docs/latest/quick-start.html) with YugabyteDB, and perform YSQL queries.
+
+## Prerequisites
+
+This tutorial assumes that you have:
+
+- YugabyteDB running. If you are new to YugabyteDB, follow the steps in [Quick start](../../../../quick-start/).
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/scala-ysql.md:51]: fix link

File: docs/content/preview/integrations/apache-spark/scala-ysql.md [hunk 5]
```diff
@@ -0,0 +1,313 @@
...
+      /_/
+
+Using Scala version 2.12.15 (OpenJDK 64-Bit Server VM, Java 1.8.0_292)
+Type in expressions to have them evaluated.
+Type :help for more information.
+scala>
+```
+
+## Set up the database
+
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/scala-ysql.md:80]: same comments as python/java rest of page

File: docs/content/preview/integrations/apache-spark/scala-ysql.md [hunk 6]
```diff
@@ -0,0 +1,313 @@
...
+    |    14.0|244487874|
+    |    19.0|256220339|
+    |     3.0|247630466|
+    |     2.0|249126085|
+    +--------+---------+
+    ```
+
+### Using spark.sql() API
+
+1. You can use the `spark.sql()` API to directly execute SQL queries from spark shell using the following code:
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/scala-ysql.md:184]: Scala shell

File: docs/content/preview/integrations/apache-spark/python-ysql.md [hunk 7]
```diff
@@ -0,0 +1,292 @@
...
+    ```
+
+    ```output
+     count
+    --------
+     100000
+    (1 row)
+    ```
+
+1. Use the `append` [SaveMode](https://spark.apache.org/docs/latest/sql-data-sources-load-save-functions.html#save-modes), to append data from `test_copy` to the `test` table as follows:
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/python-ysql.md:232]: no comma

File: docs/content/preview/integrations/apache-spark/spark-sql.md [hunk 8]
```diff
@@ -0,0 +1,199 @@
+---
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/spark-sql.md:1]: same comments as python/java

File: docs/content/preview/integrations/apache-spark/spark-sql.md [hunk 9]
```diff
@@ -0,0 +1,199 @@
...
+     ```spark
+     spark-sql> CREATE TABLE test_ref USING org.apache.spark.sql.jdbc OPTIONS (
+                         url "jdbc:yugabytedb://localhost:5433/ysql_spark_sql",
+                         dbtable "test",
+                         user "yugabyte",
+                         password "yugabyte",
+                         driver "com.yugabyte.Driver" );
+     ```
+
+You can now read and write data through the table `test_ref`.
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/spark-sql.md:104]: indent?

File: docs/content/preview/integrations/apache-spark/spark-sql.md [hunk 10]
```diff
@@ -0,0 +1,199 @@
...
+
+     ```sql
+     ysql_spark_sql=# CREATE TABLE test AS SELECT generate_series(1,100000) AS id, random(), ceil(random() * 20);
+     ```
+
+## Store and retrieve data
+
+1. Create a table `test_ref` in the `spark-sql` shell, referencing the table `test` through the configuration properties using the JDBC connector:
+
+     ```spark
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/spark-sql.md:95]: sql i think; below as well

File: docs/content/preview/integrations/apache-spark/java-ycql.md [hunk 11]
```diff
@@ -253,4 +255,4 @@ updatedRows.write().format("org.apache.spark.sql.cassandra").option("keyspace",
 
 Set to true to cause all null values to be left as unset rather than bound.
 
-For example, for `{"dl": null, "ul":"foo"}`, if `spark.cassandra.output.ignoreNulls` is specified, only the `ul` field gets written (merged) into the corresponding JSONB column, and the `dl` field is ignored.
+For example, for `{"dl": null, "ul":"foo"}`, if `spark.cassandra.output.ignoreNulls` is specified, only the `ul` field gets written (merged) into the corresponding JSONB column, and the `dl` field is ignored. -->
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/preview/integrations/apache-spark/java-ycql.md:258]: My plan is to clean up the YCQL docs too(which are pretty old) , but after addressing the YSQL ones. So wanted to keep it an easy review initially.

File: docs/content/preview/integrations/apache-spark/python-ysql.md [hunk 12]
```diff
@@ -172,9 +174,9 @@ Create a [DataFrame](https://spark.apache.org/docs/1.5.1/api/java/org/apache/spa
     +--------+---------+
     ```
 
-### Using spark.sql() API
+### Use spark.sql() API
 
-1. You can use the `spark.sql()` API to directly execute SQL queries using the following code:
+- You can use the `spark.sql()` API to directly execute SQL queries using the following code:
```
Comment (ddhodge) [inline docs/content/preview/integrations/apache-spark/python-ysql.md:179]: No I mean these can just be paragraphs; not bulleted either.

File: docs/content/preview/integrations/apache-spark/python-ycql.md [hunk 13]
```diff
@@ -1,33 +1,35 @@
 ---
```
[the remark this answers is not in the document]
Comment (aishwarya24) [reply docs/content/preview/integrations/apache-spark/python-ycql.md:1]: will create another issue to work on it in another PR.

**Pull request merged**